src/game/mario_actions_stationary.c

```diff
@@ -1039,8 +1039,12 @@ s32 act_first_person(struct MarioState *m) {
         return set_mario_action(m, ACT_IDLE, 0);
     }
 
-    if (m->floor->type == SURFACE_LOOK_UP_WARP
+#ifdef UNLOCK_ALL
+    if (m->floor != NULL && m->floor->type == SURFACE_LOOK_UP_WARP) {
+#else
+    if (m->floor != NULL && m->floor->type == SURFACE_LOOK_UP_WARP
```

Comment:
you dont need null checks with mario code since its already assumed near the beginning of mario's main loop which will force him dead if his floor doesnt exist